src/PropertyDescriber/ArrayPropertyDescriber.php

```diff
@@ -27,6 +27,22 @@ class ArrayPropertyDescriber implements PropertyDescriberInterface, ModelRegistr
      */
     public function describe(array $types, OA\Schema $property, ?array $groups = null, ?OA\Schema $schema = null, array $context = [])
     {
+        if (null === $schema) {
+            trigger_deprecation(
+                'nelmio/api-doc-bundle',
+                '4.15.0',
+                'Passing null for the $schema parameter of "PropertyDescriberInterface::describe()" is deprecated. In future versions, the $schema parameter will be made non-nullable',
+            );
+        }
+
+        if (null !== $groups) {
+            trigger_deprecation(
+                'nelmio/api-doc-bundle',
+                '4.17.0',
+                'Using the $groups parameter of "PropertyDescriberInterface::describe()" is deprecated and will be removed in a future version. Pass groups via $context[\'groups\']',
+            );
+        }
```

Comment:
Would this way of checking the parameter be better for triggering deprecations?

```suggestion
		if (null !== $groups || func_num_args() >= 3) {
            trigger_deprecation(
                'nelmio/api-doc-bundle',
                '4.17.0',
                'Using the $groups parameter of "%s" is deprecated and will be removed in a future version. Pass groups via $context[\'groups\'].', __METHOD__
            );
        }

        if (\func_num_args() < 4 || null === $schema) {
            trigger_deprecation(
                'nelmio/api-doc-bundle',
                '4.15.0',
                'The "%s()" method will have a new "OA\Schema $schema" arument not defining it is deprecated.', __METHOD__
            );

            $schema = null;
        } else {
            $schema = func_get_arg(3);
        }
```

Reply:
@DjordyKoert I'm not entirely sure that relying on the `func_num_args` count is an approach we should take here. Mainly because we deprecate _not_ defining one argument while also deprecating if the other is defined.

For example, with `func_num_args() >= 3` we wouldn't know if someone calls the method like this, which would be totally valid:
```php
$describer->describe(
    types: $types,
    property: $property,
    schema: $schema,
);
```

While `func_num_args() < 4` would not work if someone calls the method like this, even though `$schema` is not defined/`null`:
```php
    types: $types,
    property: $property,
    groups: $groups,
    context: $context,
```